rockcraft.yaml

```diff
@@ -0,0 +1,163 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+---
+
+name: charmed-ranger-rock
+base: ubuntu@22.04
+version: 2.4.0-22.04-edge
+summary: Charmed ranger ROCK OCI
+description: |
+  Apache Ranger™ is a framework to enable, monitor and
+  manage comprehensive data security across the Hadoop platform.
+license: Apache-2.0
+
+platforms:
+  amd64:
+
+parts:
+  base:
+    plugin: nil
+    source: "https://downloads.apache.org/ranger/2.4.0/apache-ranger-2.4.0.tar.gz" # yamllint disable-line
+    build-packages:
+      - build-essential
+      - curl
+      - gcc
+      - gnupg
+      - patch
+      - wget
+      - maven
+      - git
+      - openjdk-11-jdk-headless
+    build-environment:
+      - SOURCE: "https://downloads.apache.org/ranger/"
+      - VERSION: "2.4.0"
+      - TAR_FILE: "apache-ranger-2.4.0.tar.gz"
+      - JAVA_HOME: /usr/lib/jvm/java-11-openjdk-amd64
+      - RANGER_DIST: /home/ranger/dist
+    override-build: |
+      craftctl default
+
+      # make home directory
+      mkdir -p ${RANGER_DIST}
+
+      # Download the Ranger tar
+      wget "${SOURCE}/${VERSION}/${TAR_FILE}" -O ranger.tar.gz
+      wget "${SOURCE}/${VERSION}/${TAR_FILE}.sha512" -O ranger.tar.sha512
+
+      # Validate successful file download
+      sha_file=$(sed 's/^.*: //' ranger.tar.sha512)
+      hash=$(echo $sha_file | tr -d '[:space:]' | tr '[:upper:]' '[:lower:]')
+      echo "$hash  ranger.tar.gz" | sha512sum --check
+
+      # Unpack tar
+      tar -zxvf ranger.tar.gz -C ${RANGER_DIST} --strip-components 1
+      rm -rf ranger.tar.gz
+
+      # Build Ranger
+      mvn clean compile package install \
+        -DskipTests=true \
+        -Dmaven.test.skip=true \
+        -Drat.skip=true \
+        -Dpmd.skip=true \
+        -Dfindbugs.skip=true \
+        -Dspotbugs.skip=true \
+        -Dcheckstyle.skip=true \
+        -Dmaven.wagon.http.ssl.insecure=true \
+        -Dmaven.wagon.http.ssl.allowall=true \
+        -Dmaven.wagon.http.ssl.ignore.validity.dates=true
+
+      # Set up groups, users
+      groupadd ranger && \
```

Comment:
should we maybe add users at a different stage? i see data platform people put them as a separate part:
https://github.com/canonical/charmed-postgresql-rock/blob/14-22.04/rockcraft.yaml#L31

I am not sure if that is a good thing either (especially putting it on the overlay step), but to me it sounds like it should be a final step before running it

Reply:
if you don't want to change this, maybe we could at least add those other users to the ranger group at their own part?

Reply:
For this I was trying to keep as close as possible to the ranger Dockerfiles in order to make future upgrades easier. In the Dockerfiles this is all handed in the `base`: [example](https://github.com/apache/ranger/blob/52c7ef84ced065f7eaa53bcf89f4bf3d83fe4a0c/dev-support/ranger-docker/Dockerfile.ranger-base). I would be ok with moving it to their own part if this is preferred. I'm not sure I like the overlay as this comes before the build.

Reply:
Updated as discussed into the relevant parts.